src/sknnr_spatial/utils/image.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+import xarray as xr
+
+from ..image._base import ImageType, ImageWrapper
+from ..image.dataarray import DataArrayWrapper
+from ..image.dataset import DatasetWrapper
+from ..image.ndarray import NDArrayWrapper
+
+
+def is_image_type(X: ImageType) -> bool:
+    # Feature array images must have exactly 3 dimensions: (y, x, band) or (band, y, x)
+    if isinstance(X, (np.ndarray, xr.DataArray)):
+        return X.ndim == 3
+
+    # Feature Dataset images must have exactly 2 dimensions: (x, y)
+    if isinstance(X, xr.Dataset):
+        return len(X.dims) == 2
+
+    return False
```

Comment:
On both this line and line 31, I'm getting "code unreachable" hints in VSCode.  I assume it's because you've used `ImageType` as a type hint.  Do you want to use `Any` instead given that this is testing the passed type?

Reply:
You're definitely right that `is_image_type` should take `Any`.

For `get_image_wrapper`, I'm a little torn because internally we should always pass `ImageType` and using `Any` would prevent `mypy` from catching errors. At the same time, I think we need a runtime check if someone uses this incorrectly. Let me know what you think, but I would lean towards leaving the "unreachable" code with the understanding that it could be reachable if someone ignores the type annotation.

Reply:
Yep, I totally agree with you here.  I think I was mainly focused on `is_image_type` when I wrote this.